airbyte-integrations/connectors/source-freshdesk/source_freshdesk/source.py

```diff
@@ -2,11 +2,51 @@
 # Copyright (c) 2021 Airbyte, Inc., all rights reserved.
 #
 
+import logging
+import requests
+from typing import Any, List, Mapping, Optional, Tuple
+from requests.auth import AuthBase, HTTPBasicAuth
 
-from airbyte_cdk.sources.deprecated.base_source import BaseSource
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from source_freshdesk.streams import Agents, Companies, Contacts, Conversations, Groups, Roles, SatisfactionRatings, Skills, Tickets, TimeEntries
 
-from .client import Client
 
+class SourceFreshdesk(AbstractSource):
 
-class SourceFreshdesk(BaseSource):
-    client_class = Client
+    def _create_authenticator(self, api_key: str) -> AuthBase:
+        return HTTPBasicAuth(username=api_key, password="unused_with_api_key")
```

Comment:
You should provide a password. Otherwise you'll get an error like this:
```
$ python3 main.py check --config secrets/config.json                                     
{"type": "LOG", "log": {"level": "ERROR", "message": "Check failed"}}
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "'TypeError(\"__init__() missing 1 required positional argument: \\'password\\'\")'"}}
```

Reply:
@lgomezm I would suggest creating a small child class from `HTTPBasicAuth` (such as `HTTPBasicAuthNoPassword`), in which you set an empty default password. This will make this part of the code easier to read and understand that Freshdesk is using a specific auth mechanism.

Reply:
Sure. I added that small class in [abcae41](https://github.com/airbytehq/airbyte/pull/12334/commits/abcae4135576a3541a76979a09babf3a955c6b8f).